tests/mortar/poisson_matrix_based_01.cc

```diff
@@ -0,0 +1,225 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 The Lethe Authors
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception OR LGPL-2.1-or-later
+
+/**
+ * @brief PoissonOperator: test on rotated hypershell-hypershell geometry.
+ */
+
+#include <deal.II/base/conditional_ostream.h>
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_gmres.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <fstream>
+
+#include "./tests.h"
+
+using namespace dealii;
+
+
+template <int dim>
+class PoissonMatrixBased
+{
+public:
+  PoissonMatrixBased(const unsigned int fe_degree,
+                     const unsigned int mapping_degree,
+                     const unsigned int n_global_refinements,
+                     const double       radius)
+    : fe_degree(fe_degree)
+    , mapping_degree(mapping_degree)
+    , n_global_refinements(n_global_refinements)
+    , radius(radius)
+    , comm(MPI_COMM_WORLD)
+    , tria(comm)
+    , dof_handler(tria)
+    , fe(fe_degree)
+    , mapping(mapping_degree)
+    , quadrature(fe_degree + 1)
+  {}
+
+  void
+  generate_grid()
+  {
+    // generate merged grid
+    hyper_shell_with_hyper_shell(radius, tria);
+    tria.refine_global(n_global_refinements);
+  }
+
+  void
+  setup_dofs()
+  {
+    dof_handler.distribute_dofs(fe);
+
+    const IndexSet locally_relevant_dofs =
+      DoFTools::extract_locally_relevant_dofs(dof_handler);
+    constraints.reinit(dof_handler.locally_owned_dofs(), locally_relevant_dofs);
+    DoFTools::make_zero_boundary_constraints(dof_handler, 0, constraints);
+    DoFTools::make_zero_boundary_constraints(dof_handler, 3, constraints);
+    // DoFTools::make_zero_boundary_constraints(dof_handler, 1, constraints);
+    // DoFTools::make_zero_boundary_constraints(dof_handler, 2, constraints);
+
+
+
+    coupling = std::make_shared<CouplingOperator<dim, 1, double>>(
+      mapping,
+      dof_handler,
+      constraints,
+      quadrature,
+      4 * Utilities::pow(2, n_global_refinements + 1),
```

Comment:
```suggestion
      6 * Utilities::pow(2, n_global_refinements),
```

like in the other tests